### PR TITLE
HEL-15 | Add the initial Docker configuration

### PR DESCRIPTION
**Pull request by klempine**
I added an initial Docker configuration so that the application can be run using `docker-compose up`. The setup will start a PostgreSQL database and a Django backend.

I changed the `settings.py` so that most of the relevant configurations are now read from the environment variables which can be entered using `docker-compose.env.yaml`. I added an example file for reference.

I updated the requirements since I had to install `django-environ` for handling the environment variables.

I added a custom command for creating an admin user and added functionality to `docker-entrypoint.sh` so that the admin password is either read from environment or generated randomly.

I modified the existing command for creating initial page contents so that it can be run multiple times without error. I also added some new page content initialization so that the front page will display at least something instead of an error. I also added initial contents for the welcome popup.

I had to modify the `UserProfile` model so that `printer_ip` is now a `GenericIPAddress` field instead of `IPAddress`. The page content creation would not start unless I made this change.  `GenericIPAddress` should be compatible with `IPAddress` using PostgreSQL, so this shouldn't be a
problem.

I added Docker description to `README.md` + added `CONTRIBUTING.md` so that the documentation is on par with our practices.

### CODE REVIEW

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    ./manage.py create_initial_page_content
+fi
+
+# Start server
+if [[ ! -z "$@" ]]; then
+    "$@"
+elif [[ "$DEV_SERVER" = "1" ]]; then
+    python ./manage.py runserver 0.0.0.0:8080
+else
+    uwsgi --ini .prod/uwsgi.ini
```
Comment (charn) [inline docker-entrypoint.sh:42]: I think this image is missing `uwsgi` and the this uWSGI production configuration file.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+COPY --chown=appuser:appuser requirements*.txt /app/
+
+RUN apt-install.sh \
+    build-essential \
+    libpq-dev \
+    && pip install --no-cache-dir \
+    -r /app/requirements.txt \
+    #&& pip install --no-cache-dir \
+    #-r /app/requirements-prod.txt \
```
Comment (charn) [inline Dockerfile:16]: We probably should add `requirements-prod.in` which includes `uwsgi`

File: docker-entrypoint.sh (hunk 1 above)
Comment (klempine) [reply docker-entrypoint.sh:42]: Thanks for pointing this out! I'll add the missing stuff.

File: Dockerfile (hunk 2 above)
Comment (klempine) [reply Dockerfile:16]: I'll add it.

File: docker-entrypoint.sh (hunk 1 above)
Comment (klempine) [reply docker-entrypoint.sh:42]: Added the .ini file.

File: Dockerfile (hunk 2 above)
Comment (klempine) [reply Dockerfile]: Added the requirements and changed Dockerfile accordingly.

File: hkm/models/models.py [hunk 3]
```diff
@@ -62,7 +62,7 @@ class UserProfile(BaseModel):
     language = models.CharField(verbose_name=_(u'Language'), default=LANG_FI,
                                 choices=LANGUAGE_CHOICES, max_length=4)
     is_museum = models.BooleanField(verbose_name=_(u'Is museum'), default=False)
-    printer_ip = models.IPAddressField(verbose_name=u"Museum printer IP", blank=True, null=True)
```
Comment (charn) [inline hkm/models/models.py:65]: Had to check, `IPAddressField` was removed in Django 1.9 (only supported for historical migrations). So this was necessary change. :+1:

File: requirements-prod.in [hunk 4]
```diff
@@ -0,0 +1,2 @@
+uwsgi<3
+factory_boy
```
Comment (charn) [inline requirements-prod.in:2]: I would think that in production, we do not require `factory_boy`. Though it seems to also be in `requirements.in` so we don't need to include it here.

File: hkm/migrations/0037_auto_20200601_1547.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
```
Comment (charn) [inline hkm/migrations/0037_auto_20200601_1547.py:1]: Just a personal opinion, but we could rename this file to something like `0037_change_deprecated_field_type_for_printer_ip.py`

Comment (klempine) [reply hkm/migrations/0037_auto_20200601_1547.py]: Good point. I renamed the file as you suggested.

File: requirements-prod.in (hunk 4 above)
Comment (klempine) [reply requirements-prod.in]: Thanks, I removed it now from the prod requirements.

**Pull request merged**